tools/normalize_protocol_split_rows.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * This script is the first step to normalizing the  table for projects which store
+ * comma-separated values. This script splits up cells containing commas by creating new rows for
+ * each value and maintaining unique combinations of values.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Liza Levitis <llevitis.mcin@gmail.com>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+require_once 'Database.class.inc';
+require_once 'Utility.class.inc';
+
+/*
+ * Converts an array of arrays of values for different fields into
+ * an array of unique combinations of the different values.
+ */
+function cartesian($input) {
+    // filter out empty values
+    $result = array(array());
+    foreach ($input as $key => $values) {
+        $append = array();
+        foreach($result as $product) {
+            foreach($values as $item) {
+                $product[$key] = $item;
+                // remove $key with empty value so that
+                // an empty string isn't inserted into the database instead of a default NULL value
+                if ($product[$key] == "") {
+                    unset($product[$key]);
+                }
+                $append[] = $product;
+            }
+        }
+        $result = $append;
+    }
+    return $result;
+}
+
+function split_commas($table_name) {
+    $DB = \Database::singleton();
+    // Get the list of unique IDs from the table
+    $rows = $DB->pselect("SELECT * FROM ".$table_name."", array());
+    $total_commas = 0;
+    foreach ($rows as $row) {
+        $num_commas = 0;
+        foreach($row as $key => $value) {
+            if ($key == "ID") {
+                $id = $row[$key];
+            }
+            $row[$key] = explode(",", $value);
+            if (sizeof($row[$key]) > 1) {
+                $num_commas += sizeof($row[$key]) - 1;
+            }
+        }
+        $total_commas += $num_commas;
+        if ($num_commas == 0) {
+            continue;
+        }
+        $all_combinations = cartesian($row);
+        foreach ($all_combinations as $combination) {
+            unset($combination["ID"]);
+            $DB->insert($table_name, $combination);
+        }
+        // remove the row for the original ID
+        $DB->delete($table_name, array("ID" => $id));
+    }
+    if ($total_commas == 0) {
+        echo("No commas have been detected. The $table_name table has been unaltered.\n\n");
```

Comment:
Just a minor detail;
when the user runs this script; this message, together with the message below; both get outputted to the screen which is a bit confusing (see screenshot):
![messagemisleading](https://user-images.githubusercontent.com/15198379/40498674-a3ac4db2-5f4d-11e8-898d-d5c12c1395ba.png)